### PR TITLE
feat: Add initialValues prop for sign in/up to docs

### PR DESCRIPTION
**Pull request by desiprisg**

### CODE REVIEW

**Comment by desiprisg** [comment]
I will merge this once the feature is out.

File: docs/account-portal/direct-links.mdx [hunk 1]
```diff
@@ -51,4 +53,14 @@ Specify the fallback redirects under **Account Portal -> Redirects** in the Cler
   alt="Account Portal redirects"
 />
 
-</Steps>
\ No newline at end of file
+</Steps>
+
+## Prefill sign in and sign up fields
+
+In the case of direct links, the values to be used for prefilling the fields for sign in/up can be specified via the following query parameters:
```
Comment (alexisintech) [inline docs/account-portal/direct-links.mdx:60]:
```suggestion
In the case of direct links, the values to be used for pre-filling the fields for sign-in or sign-up can be specified via the following query parameters:
```

File: docs/components/authentication/sign-in.mdx [hunk 2]
```diff
@@ -157,5 +157,12 @@ All props below are optional.
         <>The full URL or path to navigate after a successful sign up.</>,
       ],
     },
+    {
+      cells: [
+        <code>initialValues</code>,
+        <code>object</code>,
+        <>The values used to prefill the sign in fields with.</>,
```
Comment (alexisintech) [inline docs/components/authentication/sign-in.mdx:164]:
```suggestion
        <>The values used to pre-fill the sign-in fields with.</>,
```

File: docs/components/authentication/sign-up.mdx [hunk 3]
```diff
@@ -107,4 +107,5 @@ All props below are optional.
 | `afterSignInUrl` | `string` | The full URL or path to navigate to after a successful sign in. |
 | `signInUrl` | `string` | Full URL or path to the sign in page. Use this property to provide the target of the 'Sign In' link that's rendered. |
 | `afterSignUpUrl` | `string` | The full URL or path to navigate after a successful sign up. |
-| `unsafeMetadata` | `object` | An object with the key and value for unsafeMetadata that will be saved to the user after sign up.<br />E.g. `{ "company": "companyID1234" }` |
\ No newline at end of file
+| `unsafeMetadata` | `object` | An object with the key and value for unsafeMetadata that will be saved to the user after sign up.<br />E.g. `{ "company": "companyID1234" }` |
+| `initialValues` | `object` | The values used to prefill the sign in fields with. |
```
Comment (alexisintech) [inline docs/components/authentication/sign-up.mdx:111]:
```suggestion
| `initialValues` | `object` | The values used to pre-fill the sign in fields with. |
```

File: docs/references/javascript/clerk/sign-in.mdx [hunk 4]
```diff
@@ -250,3 +250,4 @@ All props below are optional.
 | `afterSignInUrl` | `string` | The full URL or path to navigate after a successful sign in. |
 | `signUpUrl` | `string` | Full URL or path to the sign up page. Use this property to provide the target of the 'Sign Up' link that's rendered. |
 | `afterSignUpUrl` | `string` | The full URL or path to navigate after a successful sign up. |
+| `initialValues` | `object` | The values used to prefill the sign in fields with. |
```
Comment (alexisintech) [inline docs/references/javascript/clerk/sign-in.mdx:253]:
```suggestion
| `initialValues` | `object` | The values used to pre-fill the sign in fields with. |
```

File: docs/references/javascript/clerk/sign-up.mdx [hunk 5]
```diff
@@ -250,3 +250,4 @@ All props below are optional.
 | `afterSignInUrl?` | `string` | The full URL or path to navigate after a successful sign in. |
 | `signInUrl` | `string` | Full URL or path to the sign in page. Use this property to provide the target of the 'Sign In' link that's rendered. |
 | `afterSignUpUrl?` | `string` | The full URL or path to navigate after a successful sign up. |
+| `initialValues` | `object` | The values used to prefill the sign in fields with. |
```
Comment (alexisintech) [inline docs/references/javascript/clerk/sign-up.mdx:253]:
```suggestion
| `initialValues` | `object` | The values used to pre-fill the sign in fields with. |
```

**Review by alexisintech — CHANGES_REQUESTED** [review]

File: docs/account-portal/direct-links.mdx [hunk 6]
```diff
@@ -51,4 +53,14 @@ Specify the fallback redirects under **Account Portal -> Redirects** in the Cler
   alt="Account Portal redirects"
 />
 
-</Steps>
\ No newline at end of file
+</Steps>
+
+## Prefill sign in and sign up fields
```
Comment (alexisintech) [inline docs/account-portal/direct-links.mdx:58]: Can we add an example of how to use one of these query parameters

**Review by alexisintech — CHANGES_REQUESTED** [review]

File: docs/account-portal/direct-links.mdx (hunk 1 above)
Comment (alexisintech) [reply docs/account-portal/direct-links.mdx:60]: you're right I just checked the dictionary!! I apologize, I was following my silly autocorrect.

File: docs/account-portal/direct-links.mdx (hunk 6 above)
Comment (alexisintech) [reply docs/account-portal/direct-links.mdx:58]: @desiprisg

**Review by alexisintech — CHANGES_REQUESTED** [review]
We also need to update other documentation pages that involve these new types.
Per our slack discussion:
adding these types updates our [RedirectToSignIn](https://github.com/clerkinc/clerk-docs/blob/main/docs/components/control/redirect-to-signin.mdx/#L2) component and  component and our [RedirectToSignUp](https://github.com/clerkinc/clerk-docs/blob/main/docs/components/control/redirect-to-signup.mdx/#L2) component
as well as methods on the Clerk class; I believe specifically [this one](https://clerk.com/docs/references/javascript/clerk/redirect-methods#redirect-to-sign-in) and [this one](https://clerk.com/docs/references/javascript/clerk/redirect-methods#redirect-to-sign-up)
we also are adding initialValues as a prop to the SignInProps type - page [here](https://clerk.com/docs/references/javascript/clerk/sign-in#sign-in-props)
and to the SignUpProps type - page [here](https://clerk.com/docs/references/javascript/clerk/sign-up#sign-up-props)

File: docs/references/javascript/types/sign-in-initial-values.mdx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+The type for the `initialValues` prop that is used to pre-populate the sign-in form.
+
+## Attributes
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `emailAddress?` | `string` | The email address of the user. |
+| `username?` | `string` | The username of the user. |
+| `phoneNumber?` | `string` | The phone number of the user. |
```
Comment (alexisintech) [inline docs/references/javascript/types/sign-in-initial-values.mdx:16]: Should these have `/| undefined` like `SignUpInitialValues` has?

```suggestion
| `emailAddress?` | `string /| undefined` | The email address of the user. |
| `username?` | `string /| undefined` | The username of the user. |
| `phoneNumber?` | `string /| undefined` | The phone number of the user. |
```

**Review by alexisintech — APPROVED** [review]

File: docs/references/javascript/types/sign-in-initial-values.mdx (hunk 7 above)
Comment (desiprisg) [reply docs/references/javascript/types/sign-in-initial-values.mdx:16]: Yes, they can be `undefined`.

**Review by panteliselef — APPROVED** [review]

**Pull request merged**